File: src/main/java/no/nav/familie/integrasjoner/config/IntegrasjonConfig.java
package no.nav.familie.integrasjoner.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import no.nav.familie.http.sts.StsRestClient;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

import java.net.URI;

@Configuration
public class IntegrasjonConfig {

    @Bean
    @Autowired
    @Profile("!mock-sts")
    public StsRestClient stsRestClient(ObjectMapper objectMapper,
                                       @Value("${STS_URL}") URI stsUrl,
                                       @Value("${CREDENTIAL_USERNAME}") String stsUsername,
                                       @Value("${CREDENTIAL_PASSWORD}") String stsPassword) {

        final var stsFullUrl = URI.create(stsUrl + "?grant_type=client_credentials&scope=openid");

        return new StsRestClient(objectMapper, stsFullUrl, stsUsername, stsPassword);
    }
}

File: src/main/java/no/nav/familie/integrasjoner/config/IntegrasjonConfig.kt
package no.nav.familie.integrasjoner.config

import no.nav.familie.kontrakter.felles.jsonMapper
import no.nav.familie.restklient.sts.StsRestClient
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile
import java.net.URI

@Configuration
class IntegrasjonConfig {
    @Bean
    @Profile("!mock-sts")
    fun stsRestClient(
        @Value("\${STS_URL}") stsUrl: URI?,
        @Value("\${CREDENTIAL_USERNAME}") stsUsername: String,
        @Value("\${CREDENTIAL_PASSWORD}") stsPassword: String,
    ): StsRestClient {
        val stsFullUrl = URI.create(stsUrl.toString() + "?grant_type=client_credentials&scope=openid")

        return StsRestClient(jsonMapper, stsFullUrl, stsUsername, stsPassword, null)
    }
}
